packages/langium/src/grammar/langium-grammar-validator.ts

```diff
@@ -308,6 +309,16 @@ export class LangiumGrammarValidator {
         }
     }
 
+    checkFragmentsInTypes(grammar: ast.Grammar, accept: ValidationAcceptor): void {
```

Comment:
Please register this validation check for the `AtomType` type instead of `Grammar`, then you can write it like this:
```typescript
    checkFragmentsInTypes(type: ast.AtomType, accept: ValidationAcceptor): void {
        if (ast.isParserRule(...
    } 
```